gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/oauth/WordPressOauth.kt

```diff
@@ -1,11 +1,20 @@
 package com.gravatar.quickeditor.ui.oauth
 
-internal object WordPressOauth {
-    private const val RESPONSE_TYPE = "code"
-    private const val SCOPE = "auth"
-    private const val URL =
-        "https://public-api.wordpress.com/oauth2/authorize" +
-            "?client_id=%s&redirect_uri=%s&response_type=$RESPONSE_TYPE&scope=$SCOPE"
+import android.net.Uri
+import com.gravatar.types.Email
 
-    fun buildUrl(clientId: String, redirectUri: String) = String.format(URL, clientId, redirectUri)
+internal object WordPressOauth {
+    fun buildUrl(clientId: String, redirectUri: String, email: Email): String {
+        return Uri.Builder()
+            .scheme("https")
+            .authority("public-api.wordpress.com")
+            .appendPath("oauth2")
+            .appendPath("authorize")
+            .appendQueryParameter("client_id", clientId)
+            .appendQueryParameter("redirect_uri", redirectUri)
+            .appendQueryParameter("response_type", "code")
+            .appendQueryParameter("user_email", email.toString())
+            .build()
+            .toString()
```

Comment:
I think this is way cleaner and handles the URL encoding on to of that.

Reply:
Code looks better that way, but we need to keep the `auth` scope parameter. If we don't keep it it default to "full access to single site" which fails for users who don't have any site.

Example for the same user with no site:


`scope=auth`| no `scope` param
-|-
![Screenshot_20240903_142021](https://github.com/user-attachments/assets/c67c90d9-752b-41a4-9eba-0252abadf48f)|![Screenshot_20240903_142047](https://github.com/user-attachments/assets/d3b6cf3b-c935-4aa2-85c7-6611215ba89e)

Reply:
hmm interesting, this works just fine for me. I'm not saying we shouldn't though, will ask Lighthouse folks. 

BTW. Looks like you have an app created in the WP console. You should see Gravatar UI not WP one. That might be the difference.

Reply:
Right, I was using an old wp.com clientId/clientSecret pair. This works fine now with a gravatar.com clientId/clientSecret and the default scope. We don't need to add any scope if there is no gravatar specifics.

Reply:
I added the auth scope for now. We can update it later once we have the Gravatar scopes.

Reply:
👍 Created https://github.com/Automattic/Gravatar-SDK-Android/issues/294 to keep track of it